packages/account/src/Sections/Profile/PersonalDetails/validation.ts

```diff
@@ -0,0 +1,168 @@
+        .max(50, localize('You should enter 2-50 characters.'))
+        .matches(/^(?!.*\s{2,})[\p{L}\s'.-]{2,50}$/u, localize('Letters, spaces, periods, hyphens, apostrophes only.')),
+    last_name: Yup.string()
+        .required(localize('Last name is required.'))
+        .min(2, localize('You should enter 2-50 characters.'))
+        .max(50, localize('You should enter 2-50 characters.'))
+        .matches(/^(?!.*\s{2,})[\p{L}\s'.-]{2,50}$/u, localize('Letters, spaces, periods, hyphens, apostrophes only.')),
+    phone: Yup.string()
+        .required(localize('Phone is required.'))
+        .min(9, localize('You should enter 9-35 numbers.'))
+        .max(35, localize('You should enter 9-35 characters.'))
+        .matches(/^\+?([0-9-]+\s)*[0-9-]+$/, localize('Please enter a valid phone number (e.g. +15417541234).')),
+    address_line_1: Yup.string()
+        .trim()
+        .required(localize('First line of address is required.'))
+        .max(70, localize('Should be less than 70.'))
+        .matches(
+            /^[\p{L}\p{Nd}\s'.,:;()\u00b0@#/-]{0,70}$/u,
+            localize('Use only the following special characters: {{permitted_characters}}', {
+                permitted_characters: address_permitted_special_characters_message,
+                interpolation: { escapeValue: false },
+            })
+        ),
+    address_line_2: Yup.string()
+        .trim()
+        .max(70, localize('Should be less than 70.'))
+        .matches(
+            /^[\p{L}\p{Nd}\s'.,:;()\u00b0@#/-]{0,70}$/u,
+            localize('Use only the following special characters: {{permitted_characters}}', {
+                permitted_characters: address_permitted_special_characters_message,
+                interpolation: { escapeValue: false },
+            })
+        ),
+    address_city: Yup.string()
+        .required(localize('Town/City is required.'))
+        .max(70, localize('Should be less than 70.'))
+        .matches(
+            /^[A-Za-z]+(?:[.' -]*[A-Za-z]*){0,70}$/,
+            localize('Only letters, space, hyphen, period, and apostrophe are allowed.')
+        ),
+});
+
+export const getPersonalDetailsInitialValues = (
+    account_settings: GetSettings,
+    residence_list: ResidenceList,
+    states_list: StatesList
+) => {
+    const initialValues: GetSettings = {
+        first_name: account_settings.first_name,
+        last_name: account_settings.last_name,
+        phone: account_settings.phone,
+        date_of_birth: account_settings.date_of_birth,
+        residence: account_settings.residence,
+        address_line_1: account_settings.address_line_1,
+        address_city: account_settings.address_city,
+        address_state: '',
+        tax_identification_number: account_settings.tax_identification_number ?? '',
+        email_consent: account_settings.email_consent ?? 0,
+    };
+
+    const isGetSettingsKey = (value: string): value is keyof GetSettings => {
```

Comment:
what is `value is keyof GetSettings`?
will `(value: string):boolean` or simply `(value: string)` work?

Reply:
This is to be more specific where the `value` is string and is the `keyof GetSettings`